v3/plugins/redhat/java/0.38.0/meta.yaml

```diff
@@ -15,6 +15,10 @@ spec:
     - image: "docker.io/eclipse/che-remote-plugin-runner-java8:next"
       name: vscode-java
       memoryLimit: "1500Mi"
+      command: 
+        - sh
+        - -c
+        - /before-start.sh && ${PLUGIN_REMOTE_ENDPOINT_EXECUTABLE}
```

Comment:
I don't really like the idea of updating all our old versions of plugins, especially since it requires understanding how remote plugin runner starts up; I think the entrypoint update should either be done in the plugin runner image or by the plugin broker when it provisions a remote runtime injector.

Also, does overriding this remove the UID fixes in the generic remote runtime entrypoint?

Reply:
Preferably the plugin runner images could be updated to automatically look for and execute the injected runtime executable.

Reply:
> I think the entrypoint update should either be done in the plugin runner image or by the plugin broker when it provisions a remote runtime injector

+1

Reply:
> Preferably the plugin runner images could be updated to automatically look for and execute the injected runtime executable.

@amisevsk @davidfestal one of our goal for remote binary is ability to reuse existed rhel, ubi images without patching and rebuilding this images for CodeReady workspaces. And there a lot of VsCode extension which could work without UID fixes. But to improve this flow we really can use alternative solution. In the image with remote binary we can provide one more file - some script entrypoint.sh. Instead of running remote-binary in a straight way we will start it using this script. And this script could do some additional work: on start take a look if in the root '/' located entrypoint.sh - than start it and detach, if we don't have this file script could try to do uid fix and the end execute remote binary. 
Proposed draft content:

entrypoint.sh for remote binary
```shell
#!/bin/sh


fixUID() {
        # Ensure $HOME exists when starting
        if [ ! -d "${HOME}" ]; then
          mkdir -p "${HOME}"
        fi
        
        # Setup $PS1 for a consistent and reasonable prompt
        if [ -w "${HOME}" ] && [ ! -f "${HOME}"/.bashrc ]; then
          echo "PS1='\s-\v \w \$ '" > "${HOME}"/.bashrc
        fi
        
        # Add current (arbitrary) user to /etc/passwd
        if ! whoami 2>&1 /dev/null; then
          if [ -w /etc/passwd ]; then
            echo "${USER_NAME:-user}:x:$(id -u):0:${USER_NAME:-user} user:${HOME}:/bin/bash" >> /etc/passwd
          fi
        fi
}


if [ -f "/entrypoint.sh" ]; then
        /entrypoint.sh &
else 
        fixUID
fi

${PLUGIN_REMOTE_ENDPOINT_EXECUTABLE}
```


@amisevsk @davidfestal This solution is OK for you?

Reply:
@AndrienkoAleksandr +1 to that suggestion; I'd prefer to not have to override entrypoint on plugin runner -- ideally we could add this functionality without changing meta.yamls in the plugin registry.